logstash-core/lib/logstash/api/lib/app/service.rb

```diff
@@ -23,6 +23,10 @@ def agent
     LogStash::Instrument::Collector.instance.agent
   end
 
+  def started?
+    !@snapshot.nil? && has_counters?
```

Comment:
since we check if @snapshot is nil in has_counters?, isn't this first check redundant?